orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/appengine/AppEngineBranchFinder.groovy

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.providers.appengine
+
+import com.netflix.spinnaker.orca.pipeline.model.Pipeline
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+
+import java.util.regex.Pattern
+
+class AppEngineBranchFinder {
+  static void populateFromStage(Map operation, Stage stage) {
+    if (operation.fromTrigger && stage.execution instanceof Pipeline) {
+      def (source, project, slug, branch) = operation.branch.tokenize(":")
```

Comment:
This felt like a good idea when I wrote it. I'll just send the same information in a map under `operation.trigger` (the properties on `operation` come from the deploy description).